### ISSUE TITLE
Annoying printing of "Scale for * is already present"

### ISSUE THREAD

**Issue by jtlandis** [jtlandis/ggside#20]
I've introduced an annoying "bug" into the latest release. It isnt a breaking bug - but will definitely be a drop in quality of life for those who use `ggside` with either `facet_wrap` or `facet_grid`

``` r
library(ggside)
#> Loading required package: ggplot2
#> Registered S3 method overwritten by 'ggside':
#>   method from   
#>   +.gg   ggplot2

p <- ggplot(iris, aes(Sepal.Width, Sepal.Length, color = Species)) +
  geom_point() + 
  geom_xsidedensity(aes(y=stat(density))) +
  geom_ysidedensity(aes(x=stat(density))) +
  theme_bw()

p
```

![](https://i.imgur.com/9uZizel.png)

``` r

p + facet_wrap(~Species)
#> Scale for 'x' is already present. Adding another scale for 'x', which will
#> replace the existing scale.
#> Scale for 'x' is already present. Adding another scale for 'x', which will
#> replace the existing scale.
#> Scale for 'y' is already present. Adding another scale for 'y', which will
#> replace the existing scale.
#> Scale for 'y' is already present. Adding another scale for 'y', which will
#> replace the existing scale.
#> Scale for 'x' is already present. Adding another scale for 'x', which will
#> replace the existing scale.
#> Scale for 'x' is already present. Adding another scale for 'x', which will
#> replace the existing scale.
#> Scale for 'y' is already present. Adding another scale for 'y', which will
#> replace the existing scale.
#> Scale for 'y' is already present. Adding another scale for 'y', which will
#> replace the existing scale.
```

![](https://i.imgur.com/lvTEa0n.png)

``` r

p + facet_wrap(~Species) +
  ggside(collapse = "all")
#> Scale for 'x' is already present. Adding another scale for 'x', which will
#> replace the existing scale.
#> Scale for 'x' is already present. Adding another scale for 'x', which will
#> replace the existing scale.
#> Scale for 'y' is already present. Adding another scale for 'y', which will
#> replace the existing scale.
#> Scale for 'y' is already present. Adding another scale for 'y', which will
#> replace the existing scale.
```

![](https://i.imgur.com/sVtijbT.png)

<sup>Created on 2021-07-20 by the [reprex package](https://reprex.tidyverse.org) (v2.0.0)</sup>

**Comment by jtlandis** [comment]
The issue came sometime when trying to fix #11. The function I was using was adding too many position scales. Since this fix will be called by each 'layer' object, and each `*sideLayer` will presumedly share at least 1 axis, I only need to grab a clone of the appropriate scale.

**Comment by jtlandis** [comment]
``` r
library(dplyr)
#> 
#> Attaching package: 'dplyr'
#> The following objects are masked from 'package:stats':
#> 
#>     filter, lag
#> The following objects are masked from 'package:base':
#> 
#>     intersect, setdiff, setequal, union
library(ggside)
#> Loading required package: ggplot2
#> Registered S3 method overwritten by 'ggside':
#>   method from   
#>   +.gg   ggplot2

p <- ggplot(iris, aes(Sepal.Width, Sepal.Length, color = Species)) +
  geom_point() + 
  geom_xsidedensity(aes(y=stat(density))) +
  geom_ysidedensity(aes(x=stat(density))) +
  theme_bw()
```

``` r
p
```

![](https://i.imgur.com/W4vLPQG.png)

``` r

p + facet_wrap(~Species)
```

![](https://i.imgur.com/rtcNQes.png)

``` r

p + facet_wrap(~Species) +
  ggside(collapse = "all")
```

![](https://i.imgur.com/vQLEyvl.png)

``` r
set.seed(1234)
df <- data.frame(year = sample(2000:2020, 300, replace = T),
                 month = sample(1:12, 300, replace = T),
                 day = sample(1:28,300, replace = T),
                 temperature = rnorm(300, 70, 8)) %>%
  mutate(
    month_name = month.name[month],
    date = as.Date(sprintf("%04d-%02d-%02d",year, month, day))
  )

ggplot(df, aes(x = date, y = temperature)) +
  geom_line() +
  geom_point(aes(color = month_name)) + 
  geom_ysidehistogram() + scale_ysidex_continuous()
#> `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.
```

![](https://i.imgur.com/rOKVRyf.png)

<sup>Created on 2021-07-20 by the [reprex package](https://reprex.tidyverse.org) (v2.0.0)</sup>
